web/tests/dispatch.test.js

```diff
@@ -49,6 +49,7 @@ const realm_icon = mock_esm("../src/realm_icon");
 const realm_logo = mock_esm("../src/realm_logo");
 const realm_playground = mock_esm("../src/realm_playground");
 const reload = mock_esm("../src/reload");
+const saved_replies = mock_esm("../src/saved_replies");
```

Comment:
```suggestion
const saved_replies = zrequire("../src/saved_replies");
```

Reply:
Using `zrequire` was actually part of my suggestion in https://github.com/zulip/zulip/pull/31359#discussion_r1734034112. So basically using `zrequire` would allow you to maintain the module and its functions as they are -- So like in this case, you can actually store and get things from `saved_replies` data structure... Just like how `message_store` is being tested in here.

`mock_esm` basically mocks the module so that you can override functions as you wish... I feel verifying the contents of data structure from the event dispatch would make more sense than verifying just whether the function is called.

Reply:
This module uses `mock_esm` for basically everything, probably it's reasonable to preserve that model. But I think the `saved_replies/saved_replies_ui` split would help here.